### PR TITLE
Fixed language setting changes not requiring a restart

### PR DESCRIPTION
**Pull request by Mailaender**
A tiny step towards https://github.com/OpenRA/OpenRA/issues/10475.

### CODE REVIEW

**Review by abcdefg30 — DISMISSED** [review]
We should probably just remove this setting if it's not exposed, but meanwhile these changes won't hurt.

File: OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs [hunk 1]
```diff
@@ -319,7 +322,7 @@ Action InitDisplayPanel(Widget panel)
 
 			var restartDesc = panel.Get("RESTART_REQUIRED_DESC");
 			restartDesc.IsVisible = () => ds.Mode != OriginalGraphicsMode || ds.VideoDisplay != OriginalVideoDisplay || ds.GLProfile != OriginalGLProfile ||
-				(ds.Mode == WindowMode.Windowed && (origWidthText != windowWidth.Text || origHeightText != windowHeight.Text));
+				(ds.Mode == WindowMode.Windowed && (origWidthText != windowWidth.Text || origHeightText != windowHeight.Text || ds.Language != OriginalLanguage));
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: Won't this make it only restart in Windowed mode?

Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: You are absolutely right. I didn't notice as I always run in windowed mode.

File: OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs [hunk 2]
```diff
@@ -99,12 +101,13 @@ public SettingsLogic(Widget widget, Action onExit, ModData modData, WorldRendere
...
+					current.Graphics.Mode != OriginalGraphicsMode ||
 					current.Graphics.VideoDisplay != OriginalVideoDisplay ||
-				    current.Graphics.WindowedSize != OriginalGraphicsWindowedSize ||
+					current.Graphics.WindowedSize != OriginalGraphicsWindowedSize ||
 					current.Graphics.FullscreenSize != OriginalGraphicsFullscreenSize ||
-				    current.Graphics.GLProfile != OriginalGLProfile ||
-					current.Server.DiscoverNatDevices != OriginalServerDiscoverNatDevices)
+					current.Graphics.GLProfile != OriginalGLProfile ||
+					current.Server.DiscoverNatDevices != OriginalServerDiscoverNatDevices ||
+					current.Graphics.Language != OriginalLanguage)
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: Style nit: I'd prefer to keep the .Graphics entries grouped, so imo this should go between GLProfile and DiscoverNatDevices.

File: OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs [hunk 3]
```diff
@@ -319,7 +322,8 @@ Action InitDisplayPanel(Widget panel)
 
 			var restartDesc = panel.Get("RESTART_REQUIRED_DESC");
 			restartDesc.IsVisible = () => ds.Mode != OriginalGraphicsMode || ds.VideoDisplay != OriginalVideoDisplay || ds.GLProfile != OriginalGLProfile ||
-				(ds.Mode == WindowMode.Windowed && (origWidthText != windowWidth.Text || origHeightText != windowHeight.Text));
+				(ds.Mode == WindowMode.Windowed && (origWidthText != windowWidth.Text || origHeightText != windowHeight.Text))
+				|| ds.Language != OriginalLanguage;
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: Style nit: We have the `||` at the end of the previous line above, I'd prefer we stick to that approach here, if only for consistency.

**Comment by reaperrr** [comment]
👍 after that.

**Comment by Mailaender** [comment]
Updated for better readability.

**Comment by pchote** [comment]
> We should probably just remove this setting if it's not exposed, but meanwhile these changes won't hurt.

See #18959.

**Comment by Mailaender** [comment]
I have it enabled at @OpenHV for testing.

**Comment by pchote** [comment]
#18959 was merged, so this is no longer needed.

**Pull request closed without merging**